internal/utils/cdx.go

```diff
@@ -0,0 +1,29 @@
+package utils
+
+import (
+	cdx "github.com/CycloneDX/cyclonedx-go"
+)
+
+func CDXDiscoverComps(comps *[]*cdx.Component, comp *cdx.Component) {
```

Comment:
should we make this private? I don't find the difference between `CDXDiscoverComps` and `CDXRetrieveComps` too obvious, plus there is no documentation to understand in how they differ (without looking at the code).